### DISCUSSION TITLE
Handling data with GCPs or RPCs

### DISCUSSION THREAD

**Discussion by rbudhu** [developmentseed/titiler#283] [Q&A] [answered]
Hi:  I am trying to use TiTiler for image data that does not have CRS information in the header, but does have ground control points and rational polynomial coefficients (example: a NITF converted to a COG).  Could TiTiler be updated to use this information (rasterio support calculating transforms with GCP and RPC values)?  Can such a capability be added to the feature roadmap?  Thanks!

**Comment by vincentsarago** [comment, marked as answer]
Hi @rbudhu 
could you check if you can open your file using `rio_tiler.io.GCPCOGReader` ?

```python
from rio_tiler.io import GCPCOGReader

with GCPCOGReader("youfile") as cog:
    print(cog.info())
``` 

if this works, then you could create your own application as shown here https://developmentseed.org/titiler/examples/code/mini_cog_tiler/

but instead of using default COGReader use `GCPCOGReader` as the reader 
```python
cog = TilerFactory(reader=GCPCOGReader)
```

**Reply by rbudhu** [reply]
Awesome!  Seems to work!  Thanks!
